### ISSUE TITLE
Can't set attribute error, when trying to use the same name as a Field

### ISSUE THREAD

**Issue by rossmacarthur** [rossmacarthur/serde#41]
How to replicate:

```python
from serde import Model, field

class A(Model):
    derp = field.Field()

class B(A):
    @property
    def derp(self):
        return 'something else'

B()  # does not work
```

Exception is 

```
serde/model.py", line 284, in __init__
    setattr(self, name, value)
AttributeError: can't set attribute
```

**Comment by rossmacarthur** [comment]
I think class attributes / methods / properties with the same name as a field should take precedance, and those Fields will be dropped.

**Comment by rossmacarthur** [comment]
Resolved by #42
